File: middleoflinkedlist/soln.js
class ListNode1 {
    constructor(val, next) {
        this.val = (val===undefined ? 0 : val)
        this.next = (next===undefined ? null : next)
    }
}

class Solution1 {
    middleNode(head) {
        let slow = head
        let fast = head

        while (fast != null && fast.next != null) {
            slow = fast.next
            fast = fast.next.next
        }

        return slow
    }
}

File: middleoflinkedlist/soln.go
package middleoflinkedlist

// https://leetcode.com/problems/middle-of-the-linked-list/description/
 type ListNode struct {
	Val int
	Next *ListNode
}

func middleNode(head *ListNode) *ListNode {
	slow := head
	fast := head

	for fast != nil && fast.Next != nil {
		slow = slow.Next
		fast = fast.Next.Next
	}

	return slow
}